### PR TITLE
Inverse Laplace ∇⁻²! via kernel in ∇²!

### LINKED ISSUE #141: Inverse Laplacian
I’d need the inverse laplacian to convert vorticity to streamfunction. I’ve seen that this is available in the code, but commented out in `spectral_gradients!` as `∇⁻²!`. Why is that so? Could we make it functional again? `SpectralTransform` still has the `eigenvalues⁻¹` field that I also only used by `∇⁻²!`

`UV_from_vor!` also somehow inverts the laplacian but I don't understand how. Anyway though, I need to compute the stream function not the wind field.

### PR DESCRIPTION
**Pull request by milankl**
fixes #141

### CODE REVIEW

**Review by maximilian-gelbrecht — APPROVED** [review]
Looks good to me!

File: src/spectral_transform.jl [hunk 1]
```diff
@@ -182,8 +181,6 @@ function SpectralTransform( ::Type{NF},                     # Number format NF
 
     # EIGENVALUES (on unit sphere, hence 1/radius²-scaling is omitted)
     eigenvalues = [-l*(l+1) for l in 0:lmax+1]
-    eigenvalues⁻¹ = inv.(eigenvalues)
-    eigenvalues⁻¹[1] = 0                    # set the integration constant to 0
```
Comment (milankl) [inline src/spectral_transform.jl:186]: I just realised that we shouldn't replace `eigenvalues⁻¹` by `1/eigenvalues` as that makes the integration constant infinity. We currently don't test for this (and it's not used by the dynamical core atm) hence the tests still pass
```julia
julia> alms = LowerTriangularMatrix(randn(ComplexF32,33,32));
julia> alms[:,1] .= real.(alms[:,1]);
julia> ∇⁻²alms = copy(alms)
julia> S = SpectralTransform(alms);
julia> SpeedyWeather.∇²!(∇⁻²alms,alms,S,inverse=true)
33×32 LowerTriangularMatrix{ComplexF32}:
   -Inf+NaN*im                0.0+0.0im          …          0.0+0.0im
    -0.369622-0.0im      0.199208-0.408254im                0.0+0.0im
...
```

**Comment by milankl** [comment]
Don't worry, I'll resolve that, just don't merge it yet. Enjoy your vacation 🌴

**Comment by milankl** [comment]
Also @maximilian-gelbrecht, a little change I made in `∇²!` and `∇⁻²!` is that previously it excluded the the last row = highest degree l. This is because scalar quantities have lmax==mmax, but vector quantities use one more degree due to the recursion in the meridional gradient. I changed this to include all degrees and orders, to have a generally valid (inverse) Laplace operator (the higest degree l otherwise wouldn't be touched, and values may carry over which I don't like). I believe this does not have any unforeseen consequences but in case it does, let me know.

**Comment by maximilian-gelbrecht** [comment]
Great, thanks @milankl

**Pull request merged**